File: parse_questions.py
import re
import json

with open('/Users/Sean-Work/Desktop/speech-activities-html/activities/reading/late-winter-early-spring-reading.html', 'r', encoding='utf-8') as f:
    text = f.read()

# Find the stories array
match = re.search(r'const stories = (\[.*?\]);\s*<\/script>', text, re.DOTALL)
if match:
    stories_json = match.group(1)
    # The JSON might have some slight JS syntax (like single quotes or trailing commas), 
    # but looking at the snippet, it looks like valid JSON.
    try:
        stories = json.loads(stories_json)
        with open('questions_dump.txt', 'w', encoding='utf-8') as out:
            for story in stories:
                out.write(f"=== Story: {story.get('title')} ===\n")
                for version_name, version_data in story.get('versions', {}).items():
                    out.write(f"  -- Version: {version_name} --\n")
                    questions = version_data.get('questions', {})
                    for q_type, q_list in questions.items():
                        out.write(f"    - Type: {q_type}\n")
                        for idx, q in enumerate(q_list):
                            out.write(f"      Q{idx+1}: {q.get('prompt')}\n")
                            for c_idx, choice in enumerate(q.get('choices', [])):
                                mark = "*" if choice.get("isCorrect") else " "
                                out.write(f"        [{mark}] {choice.get('text')}\n")
                    out.write("\n")
        print("Successfully dumped to questions_dump.txt")
    except json.JSONDecodeError as e:
        print("JSON parse error", e)
else:
    print("Could not find stories JSON in the file.")

File: parse_questions.js
const fs = require('fs');

const text = fs.readFileSync('/Users/Sean-Work/Desktop/speech-activities-html/activities/reading/late-winter-early-spring-reading.html', 'utf8');

const match = text.match(/const stories = (\[[\s\S]*?\]);\s*<\/script>/);
if (match) {
    const storiesJSON = match[1];
    
    // Evaluate the matching string as JavaScript
    let stories;
    try {
        stories = eval(storiesJSON);
    } catch (e) {
        console.error("Failed to eval stories:", e);
        process.exit(1);
    }

    let output = '';
    
    stories.forEach(story => {
        output += `\n=== Story: ${story.title} ===\n`;
        const versions = story.versions;
        for (const [vName, vData] of Object.entries(versions)) {
            output += `\n  -- Version: ${vName} --\n`;
            if (vData.questions) {
                for (const [qType, qList] of Object.entries(vData.questions)) {
                    output += `    - Type: ${qType}\n`;
                    qList.forEach((q, idx) => {
                        output += `      Q${idx+1}: ${q.prompt}\n`;
                        q.choices.forEach(c => {
                            const mark = c.isCorrect ? '*' : ' ';
                            output += `        [${mark}] ${c.text}\n`;
                        });
                    });
                }
            }
        }
    });
    
    fs.writeFileSync('questions_dump.txt', output);
    console.log("Successfully dumped to questions_dump.txt");
} else {
    console.log("No match found");
}
